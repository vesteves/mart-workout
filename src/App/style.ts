import styled from "styled-components";
import { Link as LinkReact } from "react-router-dom";

export const Wrapper = styled.div`
  background: #333;
`;

export const Container = styled.div`
  margin: 0 20px;
`;

export const Link = styled(LinkReact)`
  text-decoration: none;
  color: #fff;
`;
