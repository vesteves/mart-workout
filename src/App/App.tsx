import { Routes, Route, Link } from "react-router-dom";
import Workouts from "./Workouts/Workouts";
import Home from "./Home/Home";
import Workout from "./Workout/Workout";
import * as S from "./style";

export const App = () => {
  return (
    <S.Wrapper>
      <S.Container>
        <div>
          <S.Link to="/">Home</S.Link>
        </div>
        <div>
          <S.Link to="/workouts">Workouts</S.Link>
        </div>
        <hr />
      </S.Container>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/workouts" element={<Workouts />} />
        <Route path="/workouts/:id" element={<Workout />} />
      </Routes>
    </S.Wrapper>
  );
};

export default App;
