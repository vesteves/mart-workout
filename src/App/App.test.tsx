import { render } from "@testing-library/react";
import App from "./App";
import { BrowserRouter } from "react-router-dom";

test("should match to snapshot", () => {
  const { container } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
