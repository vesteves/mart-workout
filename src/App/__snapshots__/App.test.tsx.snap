// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match to snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ beJljN"
  >
    <div
      class="sc-gsnTZi kPWacB"
    >
      <div>
        <a
          class="sc-dkzDqf cNwmzd"
          href="/"
        >
          Home
        </a>
      </div>
      <div>
        <a
          class="sc-dkzDqf cNwmzd"
          href="/workouts"
        >
          Workouts
        </a>
      </div>
      <hr />
    </div>
    <div
      class="sc-idiyUo jDyUpd"
    >
      <div
        class="sc-gsnTZi sc-dIouRR kPWacB"
      >
        <h2>
          Home
        </h2>
        <p>
          Developer: Vitor Esteves
        </p>
        <p>
          Status: Ready to be hired!
        </p>
      </div>
    </div>
  </div>
</div>
`;
